### PR TITLE
adding tidyhydat blog post

### PR DESCRIPTION
**Pull request by boshek**
@sckott @stefaniebutland - for review

### CODE REVIEW

File: content/blog/2018-01-23-tidyhydat.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
+---
+slug: tidyhydat
+title: "5 Things I Learned Making a Package to Work with Hydrometric Data in R"
+package_version: 0.3.2
+authors:
+- name: Sam Albers
+  url: https://github.com/bcgov
+date: 2018-01-23
```
Comment (stefaniebutland) [inline content/blog/2018-01-23-tidyhydat.md]: date: 2018-01-16

File: content/blog/2018-01-23-tidyhydat.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+`tidyhydat` is a new rOpenSci R package that provides a standard method of accessing [Environment and Climate Change Canada's](https://www.canada.ca/en/environment-climate-change.html) [Water Survey of Canada](https://wateroffice.ec.gc.ca/index_e.html) data sources (WSC) using a consistent and easy to use interface that employs [tidy data principles](https://cran.r-project.org/web/packages/tidyr/vignettes/tidy-data.html). I developed the package because I was repeating the [same steps](https://twitter.com/drob/status/611885584584441856?ref_src=twsrc%5Etfw&ref_url=http%3A%2F%2Ftinyheero.github.io%2Fjekyll%2Fupdate%2F2015%2F07%2F26%2Fmaking-your-first-R-package.html) to [import and tidy](http://r4ds.had.co.nz/explore-intro.html) Water Survey of Canada data in R. Many useRs will read this post and think "I’d know how to do that". To you I tip my hat. To those who read this and think "huh, never knew that", you are the target audience. In this post, I'll focus on 5 things that I learned  during the `tidyhydat` rOpenSci review, provide a brief introduction on how the package works and provide an example use case.
```
Comment (stefaniebutland) [inline content/blog/2018-01-23-tidyhydat.md]: you spell out "Water Survey of Canada" after defining abbreviation. decide which to do throughout post

File: content/blog/2018-01-23-tidyhydat.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+# Acknowledgements
+
+I would like to thank rOpenSci for the opportunity to have the `tidyhydat` package reviewed. In particular I would like to thank [SCott Chamberlain](https://github.com/sckott), [Laura DeCicco](https://github.com/ldecicco-USGS) and [Luke Winslow](https://github.com/lawinslow) for their time and efforts in improving the package. Hopefully this post has outlined the ways in which their contributions were useful. More generally `tidyhydat` offers another piece of evidence that the rOpenSci package review process consistently yields better results and returns more user friendly products that facilitate open science.
```
Comment (stefaniebutland) [inline content/blog/2018-01-23-tidyhydat.md]: Scott (fix the C)

File: content/blog/2018-01-23-tidyhydat.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+       subtitle = "Correlation of daily flows.")
+```
+
+![](../../themes/ropensci/static/img/blog-images/2018-01-23-tidyhydat/sp_plt-1.png)<!-- -->
+
+# What does the plot tell us
+Our original objective was to assess if the `08MH024` station was a **unique** station. This type of plot provides a clear pattern of hydrologic linkages. Flows at `08MH024` are more similar to stations at higher elevations (e.g. `08MH016`) than lower elevation tributaries (e.g. `08MH002`). This likely reflects the importance of upstream hydrologic process. The Fraser River, the largest river in British Columbia drains a huge area (228000 km^2^) where the flows in the majority of contributing upstream tributaries are driven by snow melt. This station, however, is located near the coastal environment where river flow is driven more by rainfall. This helps us connect upstream and downstream processes and evaluates the relative contributions to flow in the main river. A visual tool, such as the one illustrated here, provides a clear example of using the capabilities of R after bringing in a rich hydrometric data set, and  highlights a reproducible hydrological workflow in R.
```
Comment (stefaniebutland) [inline content/blog/2018-01-23-tidyhydat.md]: (228000 km^2^) superscript doesn't render in the preview. Any ideas @sckott ?

File: content/blog/2018-01-23-tidyhydat.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+
+## Installation
+<img src="../../themes/ropensci/static/img/blog-images/2018-01-23-tidyhydat/tidyhydat_large.png" alt="Drawing" style="width: 200px;"/>
```
Comment (stefaniebutland) [inline content/blog/2018-01-23-tidyhydat.md]: I think you can make path: /img/blog-images/2018-01-23-tidyhydat/tidyhydat_large.png. Same for other image

With change of post date, you'll need to change date of dir for images too

File: content/blog/2018-01-23-tidyhydat.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+## 2. Where to store external data
+
+The single biggest feature of `tidyhydat` that caused the most headaches and learning opportunities for me is the fact that much of it's capabilities rely on an external `.sqlite3` data: HYDAT. HYDAT is the WSC's quarterly published [archive of hydrometric data](http://collaboration.cmc.ec.gc.ca/cmc/hydrometrics/www/). Some stations data range extends into the 19^th^ century. It is, however, rather messy data and in a format (`.sqlite3`) that is likely unfamiliar to many useRs accustomed to flat data files like `.csv`, `.txt` or `.xlsx`.
```
Comment (stefaniebutland) [inline content/blog/2018-01-23-tidyhydat.md]: "its" no apostrophe

File: content/blog/2018-01-23-tidyhydat.md [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+<img src="../../themes/ropensci/static/img/blog-images/2018-01-23-tidyhydat/tidyhydat_large.png" alt="Drawing" style="width: 200px;"/>
+
+`tidyhydat` can be installed from CRAN by the usual manner:
+
+```r
+install.packages("tidyhydat")
+```
+
+## 1. Documentation
+A second and third set of eyes, in the form of rOpenSci reviewers, on the tidyhydat documentation added to the quality of the package immensely. It is so easy to write documentation that makes sense when you are writing the code, and makes zero sense to anyone else. Having reviewers review documentation and politely tap you on the shoulder to say "this is barely English" is the clear line between a package that is useful to only you and a package that is useful to others. From adding examples to each and every function, to suggestions that units be included in function documentation, all provide a more clear pathway for a potential new user to navigate the package.
```
Comment (stefaniebutland) [inline content/blog/2018-01-23-tidyhydat.md]: `tidyhydat` in backticks in this para

File: content/blog/2018-01-23-tidyhydat.md [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+
+## 5. An exercise in reproducible science.
+`tidyhydat` was originally developed to save time accessing data. Instead of frequently opening database connections, making approximately the same query, forgetting to close the database connection, trying to reopen it, cursing R and generally blaming all the wrong people, I developed a package.  However, after using the package it has become clear that it has a much deeper and useful feature &mdash; it is a tool that facilitates reproducible science and can provide reproducible workflows that support science-based decisions around water flow. It is now possible to set up code-based workflows that are re-runnable with more current data, different stations or regions over a range of time scales. The importance of this cannot be overstated and will be the focus of the remainder of this post. 
+
+# A reproducible work flow
```
Comment (stefaniebutland) [inline content/blog/2018-01-23-tidyhydat.md]: "how unique is this station to others" is missing word "compared"

File: content/blog/2018-01-23-tidyhydat.md [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+To illustrate a reproducible workflow facilitated by `tidyhydat` and made possible by the capabilities of R, we can ask a simple question related to a hydrometric station: how unique is this station to others in the same watershed? In this example we will consider the FRASER RIVER AT MISSION station. Records extend over many years and we can compare an overlapping record using correlation to develop a measure of **uniqueness** for this station. But first we need to engage in some preamble:
+
+## Required packages
+A portion of the package audience is likely those with little or no experience in R so an example of the type of analysis possible with tidyhydat is helpful. For our purposes here, we need to install a whole other slew of packages. The packages needed for this analysis mostly fall into either the `tidyverse` collection of packages or the `igraph` family of packages. These packages can be installed using:
```
Comment (stefaniebutland) [inline content/blog/2018-01-23-tidyhydat.md]: - put tidyhydat in backticks here.

File: content/blog/2018-01-23-tidyhydat.md (hunk 4 above)
Comment (stefaniebutland) [inline content/blog/2018-01-23-tidyhydat.md]: You don't say exactly whether or not it's unique

File: content/blog/2018-01-23-tidyhydat.md [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+
+## Installation
+<img src="../../themes/ropensci/static/img/blog-images/2018-01-23-tidyhydat/tidyhydat_large.png" alt="Drawing" style="width: 200px;"/>
+
+`tidyhydat` can be installed from CRAN by the usual manner:
+
+```r
+install.packages("tidyhydat")
+```
+
```
Comment (stefaniebutland) [inline content/blog/2018-01-23-tidyhydat.md]: I'm wondering if here, you could have heading # Five things

File: content/blog/2018-01-23-tidyhydat.md [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+`tidyhydat` is a new rOpenSci R package that provides a standard method of accessing [Environment and Climate Change Canada's](https://www.canada.ca/en/environment-climate-change.html) [Water Survey of Canada](https://wateroffice.ec.gc.ca/index_e.html) data sources (WSC) using a consistent and easy to use interface that employs [tidy data principles](https://cran.r-project.org/web/packages/tidyr/vignettes/tidy-data.html). I developed the package because I was repeating the [same steps](https://twitter.com/drob/status/611885584584441856?ref_src=twsrc%5Etfw&ref_url=http%3A%2F%2Ftinyheero.github.io%2Fjekyll%2Fupdate%2F2015%2F07%2F26%2Fmaking-your-first-R-package.html) to [import and tidy](http://r4ds.had.co.nz/explore-intro.html) Water Survey of Canada data in R. Many useRs will read this post and think "I’d know how to do that". To you I tip my hat. To those who read this and think "huh, never knew that", you are the target audience. In this post, I'll focus on 5 things that I learned  during the `tidyhydat` rOpenSci review, provide a brief introduction on how the package works and provide an example use case. 
+
+## Installation
```
Comment (stefaniebutland) [inline content/blog/2018-01-23-tidyhydat.md]: Could you do without this heading and just say "tidyhydat can be installed ..."? The little code block will draw attention to this bit anyway without having a heading for one line of text

File: content/blog/2018-01-23-tidyhydat.md (hunk 3 above)
Comment (boshek) [reply content/blog/2018-01-23-tidyhydat.md]: done

**Comment by boshek** [comment]
I may have done this incorrectly. I made the changes and just pushed them up. I agree with all of @stefaniebutland and tried to address each comment.

**Comment by stefaniebutland** [comment]
You did it right. Changes look good. Scott will review Monday (away rn)

Thanks for the nice words Sam 😊. Happy weekend

**Comment by boshek** [comment]
@sckott Struggling to center the logo. Maybe it doesn't matter and looks fine the way it is. Also have not managed to the resolve the superscript issue. Otherwise I am done editing.

**Comment by sckott** [comment]
having a look now

**Comment by sckott** [comment]
@boshek 

- for centering logo, you can do `<center>Drawing</center>`
- let's move all headers to one smaller (so add another `#`) for everyone of them - they are quite large right now.
- for images, use `/img/blog-images/ etcetera` for them to show up correctly
- for superscripts i'm not sure how to make that work. can maybe just do `19th` century as a compromise

**Comment by boshek** [comment]
@sckott 
* Headers/Centering -> yep
* Which images are not displaying correctly?
* I'm fine with 19th. Are we also ok with km2 instead of km<sup>2</sup>?

**Comment by sckott** [comment]
Both images

Hmm, yeah I guess km2 doesn't make as much sense. I'm not sure how to fix it.

**Comment by boshek** [comment]
@sckott 
* Fixed headers
* Fixed image links - displays fine on my machine (windows 7, chrome)
* Changed to square km. A bit hacky but since this is the only instance, I don't mind.

**Comment by sckott** [comment]
think it's good to go now @stefaniebutland

**Comment by stefaniebutland** [comment]
Thanks @sckott and @boshek. This will be posted this evening (= Jan 16th in some timezones) and will tweet from rOpenSci Tues AM 😄

**Pull request merged**